src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -747,6 +803,33 @@ def _on_acquire_overview(self, _):
         if das:
             self._tab.load_overview_data(das)
 
+    def _on_tdct(self, _):
+        """
+        called when the button "TDCT" is pressed
+        """
+        for stream in self._tab_data.main.currentFeature.value.streams.value:
+            if isinstance(stream, StaticFluoStream) and getattr(stream, "zIndex", None):
+                z_stack = True
+                self.txt_tdct.SetLabel("Interpolation of Z stacks .... \nMay take a while")
+                wx.CallAfter(self._on_close_dialog, z_stack)
+                return  # Prevent continuing execution here
+
+    def _on_close_dialog(self, z_stack):
```

Comment:
```suggestion
    def _on_close_dialog(self, z_stack, current_feature):
```

Maybe pass the current feature as an arg